logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -341,8 +342,27 @@ sealed interface Message {
         }
     }
 
-    enum class Status {
-        PENDING, SENT, READ, FAILED, FAILED_REMOTELY
+    sealed class Status {
+
+        object Pending : Status()
+        object Sent : Status()
+
+        object Delivered : Status()
+
+        data class Read(val readCount: Long) : Status()
+
+        object Failed : Status()
+
+        object FailedRemotely : Status()
+
+        override fun toString(): String = when (this) {
```

Comment:
Is this for logic or logging purposes? There are better ways in the code base to log a sealed class with associated value like `Read`. See for example `User` in Event.kt

Reply:
This is not really intent of this PR, if this is something pressing I can raise another PR after this one is merged

Reply:
Anyways let me adjust it

Reply:
done, let me know if t hat is what u meant